src/app/clusters/mode-base-server/mode-base-server.cpp

```diff
@@ -0,0 +1,489 @@
+        {
+            return Protocols::InteractionModel::Status::ConstraintError;
+        }
+    }
+    DataModel::Nullable<uint8_t> oldStartUpMode = mStartUpMode;
+    mStartUpMode                                = aNewStartUpMode;
+    if (mStartUpMode != oldStartUpMode)
+    {
+        // Write new value to persistent storage.
+        ConcreteAttributePath path = ConcreteAttributePath(mEndpointId, mClusterId, Attributes::StartUpMode::Id);
+        GetAttributePersistenceProvider()->WriteScalarValue(path, mStartUpMode);
+        MatterReportingAttributeChangeCallback(path);
+    }
+    return Protocols::InteractionModel::Status::Success;
+}
+
+Status Instance::UpdateOnMode(DataModel::Nullable<uint8_t> aNewOnMode)
+{
+    if (!aNewOnMode.IsNull())
+    {
+        if (!IsSupportedMode(aNewOnMode.Value()))
+        {
+            return Protocols::InteractionModel::Status::ConstraintError;
+        }
+    }
+    DataModel::Nullable<uint8_t> oldOnMode = mOnMode;
+    mOnMode                                = aNewOnMode;
+    if (mOnMode != oldOnMode)
+    {
+        // Write new value to persistent storage.
+        ConcreteAttributePath path = ConcreteAttributePath(mEndpointId, mClusterId, Attributes::OnMode::Id);
+        GetAttributePersistenceProvider()->WriteScalarValue(path, mOnMode);
+        MatterReportingAttributeChangeCallback(path);
+    }
+    return Protocols::InteractionModel::Status::Success;
+}
+
+DataModel::Nullable<uint8_t> Instance::GetStartUpMode() const
+{
+    return mStartUpMode;
+}
+
+DataModel::Nullable<uint8_t> Instance::GetOnMode() const
+{
+    return mOnMode;
+}
+
+uint8_t Instance::GetCurrentMode() const
+{
+    return mCurrentMode;
+}
+
+bool Instance::IsSupportedMode(uint8_t modeValue)
+{
+    uint8_t value;
+    for (uint8_t i = 0; mDelegate->GetModeValueByIndex(i, value) != CHIP_ERROR_PROVIDER_LIST_EXHAUSTED; i++)
+    {
+        if (value == modeValue)
+        {
+            return true;
+        }
+    }
+    ChipLogDetail(Zcl, "Cannot find a mode with value %u", modeValue);
+    return false;
+}
+
+Instance::Instance(Delegate * aDelegate, EndpointId aEndpointId, ClusterId aClusterId, uint32_t aFeature) :
+    CommandHandlerInterface(Optional<EndpointId>(aEndpointId), aClusterId),
+    AttributeAccessInterface(Optional<EndpointId>(aEndpointId), aClusterId), mDelegate(aDelegate), mEndpointId(aEndpointId),
+    mClusterId(aClusterId),
+    mCurrentMode(0), // This is a temporary value and may not be valid. We will change this to the value of the first
+                     // mode in the list at the start of the Init function to ensure that it represents a valid mode.
+    mFeature(aFeature)
+{
+    mDelegate->SetInstance(this);
+}
+
+Instance::~Instance()
+{
+    ModeBaseAliasesInstances.erase(this);
+    chip::app::InteractionModelEngine::GetInstance()->UnregisterCommandHandler(this);
+    // todo unregister the AttributeAccessOverride. Waiting for issue #28072.
```

Comment:
That issue is fixed now.  Up to you whether you want to add the call here or in a followup, just make sure it happens, please.

Reply:
Nice! Added [here](https://github.com/project-chip/connectedhomeip/pull/27504/commits/cc6458706298fa22d5918a294ee952948a1e12fc).